src/main/java/com/spotify/reaper/resources/ClusterResource.java

```diff
@@ -71,36 +68,35 @@ public Response getClusterList() {
 
   @GET
   @Path("/{cluster_name}")
-  public Response getCluster(@PathParam("cluster_name") String clusterName) {
+  public Response getCluster(
+      @PathParam("cluster_name") String clusterName,
+      @QueryParam("limit") Optional<Integer> limit) {
     LOG.info("get cluster called with cluster_name: {}", clusterName);
-    Optional<Cluster> cluster = context.storage.getCluster(clusterName);
-    if (cluster.isPresent()) {
-      return viewCluster(cluster.get(), Optional.<URI>absent());
-    } else {
-      return Response.status(Response.Status.NOT_FOUND)
-          .entity("cluster with name \"" + clusterName + "\" not found").build();
-    }
+    return viewCluster(clusterName, limit, Optional.<URI>absent());
   }
 
-  @GET
-  @Path("/{cluster_name}/{keyspace_name}")
-  public Response getCluster(@PathParam("cluster_name") String clusterName,
-                             @PathParam("keyspace_name") String keyspaceName) {
-    LOG.info("get cluster/keyspace called with cluster_name: {}, and keyspace_name: {}",
-             clusterName, keyspaceName);
-    Optional<Cluster> cluster = context.storage.getCluster(clusterName);
-    if (cluster.isPresent()) {
-      return viewKeyspace(cluster.get(), keyspaceName);
-    } else {
+  private Response viewCluster(String clusterName, Optional<Integer> limit,
+      Optional<URI> createdURI) {
+    Collection<RepairRunStatus> view =
+        context.storage.getClusterRunStatuses(clusterName, limit.or(10));
+
+    if (view == null) {
       return Response.status(Response.Status.NOT_FOUND)
           .entity("cluster with name \"" + clusterName + "\" not found").build();
+    } else if (createdURI.isPresent()) {
+      return Response.created(createdURI.get())
+          .entity(view).build();
+    } else {
+      return Response.ok()
+          .entity(view).build();
     }
   }
 
   @POST
   public Response addCluster(
       @Context UriInfo uriInfo,
-      @QueryParam("seedHost") Optional<String> seedHost) {
+      @QueryParam("seedHost") Optional<String> seedHost,
+      @QueryParam("limit") Optional<Integer> limit) {
```

Comment:
Why do you want add limit for adding a cluster, i.e. the returned new cluster will always be empty, and I find this only confusing for the user.